test/e2e/azure_test.go

```diff
@@ -16,228 +18,275 @@ limitations under the License.
-// 			},
-// 			ProviderSpec: capi.ProviderSpec{
-// 				Value: providerSpecValue,
-// 			},
-// 		},
-// 	}
-
-// 	return cluster
-// }
-
-// func makeMachine(name, clusterName, role, k8sVersion string) *capi.Machine {
-// 	var instanceRole string
-// 	machineVersionInfo := capi.MachineVersionInfo{
-// 		Kubelet: k8sVersion,
-// 	}
-
-// 	switch role {
-// 	case "control-plane":
-// 		instanceRole = "control-plane.cluster-api-provider-azure.sigs.k8s.io"
-// 		machineVersionInfo.ControlPlane = k8sVersion
-// 	case "node":
-// 		instanceRole = "nodes.cluster-api-provider-azure.sigs.k8s.io"
-// 	}
-// 	Expect(instanceRole).ToNot(BeEmpty())
-
-// 	providerSpecValue, err := capz.EncodeMachineSpec(&capz.AzureMachineProviderSpec{
-// 		// TODO: Determine bare minimum machine spec to define here
-// 		VMSize: "Standard_B2ms",
-// 	})
-// 	Expect(err).To(BeNil())
-
-// 	machine := &capi.Machine{
-// 		ObjectMeta: metav1.ObjectMeta{
-// 			Name: name,
-// 			Labels: map[string]string{
-// 				capi.MachineClusterLabelName: clusterName,
-// 				"set":                        role,
-// 			},
-// 		},
-// 		Spec: capi.MachineSpec{
-// 			Versions: machineVersionInfo,
-// 			ProviderSpec: capi.ProviderSpec{
-// 				Value: providerSpecValue,
-// 			},
-// 		},
-// 	}
-
-// 	return machine
-// }
-
-// func createNamespace(client kubernetes.Interface, namespace string) {
-// 	_, err := client.CoreV1().Namespaces().Create(&corev1.Namespace{
-// 		ObjectMeta: metav1.ObjectMeta{
-// 			Name: namespace,
-// 		},
-// 	})
-// 	Expect(err).To(BeNil())
-// }
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstrapv1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
+	kubeadmv1beta1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/kubeadm/v1beta1"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-azure/test/e2e/framework"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+var _ = Describe("CAPZ e2e tests", func() {
+	Describe("Cluster creation", func() {
+
+		clusterGen := &ClusterGenerator{}
+		nodeGen := &NodeGenerator{}
+
+		Context("Create one controlplane cluster", func() {
+			It("Should create a single node cluster and delete it before exiting", func() {
+				cluster, infraCluster := clusterGen.GenerateCluster(namespace)
+				node := nodeGen.GenerateNode(cluster.GetName())
+				OneNodeCluster(&OneNodeClusterInput{
+					Management:    mgmt,
+					Cluster:       cluster,
+					InfraCluster:  infraCluster,
+					Node:          node,
+					CreateTimeout: 60 * time.Minute,
+				})
+				framework.CleanUp(&framework.CleanUpInput{
+					Management: mgmt,
+					Cluster:    cluster,
+				})
+			})
+		})
+
+		// TODO: Deploy multiple Control Plane
```

Comment:
let's open good first issues for each of these

Reply:
listed [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/339#issuecomment-564200720)